SI/Sem.06/Pract.06/Task03.cpp

```diff
@@ -0,0 +1,26 @@
+#include <iostream>
+using namespace std;
+int whereIsTheFirstRightOne(int num)
+{
+	int rem = num % 2;
+	int i = 0;
+	while(num != 0)
+	{
+		if ( rem == 1)
+		{
+			return i;
+		}
+		i++;
+		num >>= 1;
+		rem = num % 2;
+	}
+}
+int main()
+{
+	cout << "Enter a number: ";
+	int num;
+	cin >> num;
+	int n = whereIsTheFirstRightOne(num);
+	cout << (num&~(1 << n));
```

Comment:
Again , 1 function to find the last bit AND switch it.
The solution is too complex, think of an easier way.